### ISSUE TITLE
mongodb_exporter crashes when mongo is not available

### ISSUE THREAD

**Issue by ralberto** [percona/mongodb_exporter#50]
mongodb_exporter crashes when mongodb is not available. Here's the error:

```
E0228 11:10:46.481870    7005 connection.go:27] Cannot connect to server using url mongodb://localhost:27017: no reachable servers`
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x67890e]

goroutine 26 [running]:
panic(0x7828e0, 0xc42000c090)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
gopkg.in/mgo%2ev2.(*Session).Close(0x0)
	/home/rui/Projects/golang/src/gopkg.in/mgo.v2/session.go:1612 +0x2e
github.com/percona/mongodb_exporter/collector.(*MongodbCollector).Describe(0xc42016e0a0, 0xc420062b40)
	/home/rui/Projects/golang/src/github.com/percona/mongodb_exporter/collector/mongodb_collector.go:47 +0xef
github.com/prometheus/client_golang/prometheus.(*registry).Register.func1(0x984c80, 0xc42016e0a0, 0xc420062b40)
	/home/rui/Projects/golang/src/github.com/prometheus/client_golang/prometheus/registry.go:218 +0x3b
created by github.com/prometheus/client_golang/prometheus.(*registry).Register
	/home/rui/Projects/golang/src/github.com/prometheus/client_golang/prometheus/registry.go:220 +0xc7
```

**Comment by timvaillancourt** [comment]
Thanks for this error report @ralberto. This is fixed in PR: #55. If you have time, please test this branch.

**Comment by timvaillancourt** [comment]
I believe #55 resolves this. If mongodb is restarted after the exporter has started, we will now timeout after 5 seconds instead of crash:

-logtostderr output (restart of mongod is in the middle of output):
```
I0310 17:32:57.192951   19271 mongodb_collector.go:64] Connected to: mongodb://****:****@localhost:27017 (node type: replset, server version: 3.2.11-3.1)
I0310 17:32:57.192974   19271 mongodb_collector.go:96] Collecting Server Status
I0310 17:32:57.195016   19271 mongodb_collector.go:106] Collecting Replset Status
I0310 17:32:57.196049   19271 mongodb_collector.go:112] Collecting Replset Oplog Status

E0310 17:33:03.969308   19271 connection.go:42] Cannot connect to server using url mongodb://****:****@localhost:27017: no reachable servers

E0310 17:33:10.535670   19271 connection.go:42] Cannot connect to server using url mongodb://****:****@localhost:27017: no reachable servers

E0310 17:33:17.144736   19271 connection.go:42] Cannot connect to server using url mongodb://****:****@localhost:27017: no reachable servers

I0310 17:33:19.210740   19271 mongodb_collector.go:64] Connected to: mongodb://****:****@localhost:27017 (node type: replset, server version: 3.2.11-3.1)
I0310 17:33:19.210782   19271 mongodb_collector.go:96] Collecting Server Status
I0310 17:33:19.214265   19271 mongodb_collector.go:106] Collecting Replset Status
I0310 17:33:19.214795   19271 mongodb_collector.go:112] Collecting Replset Oplog Status
```

If mongodb is unavailable on startup, we will still crash/exit (at least for now).

Please re-open if any issues are noticed.
